plugins/labormanager/labormanager.cpp

```diff
@@ -1967,13 +1998,14 @@ class AutoLaborManager {
                 if (l == df::unit_labor::NONE)
                     continue;
 
-                set_labor(*d, l,
-                    (l >= df::unit_labor::HAUL_STONE && l <= df::unit_labor::HAUL_ANIMALS) ||
-                    l == df::unit_labor::CLEAN ||
-                    l == df::unit_labor::HAUL_WATER ||
-                    l == df::unit_labor::REMOVE_CONSTRUCTION ||
-                    l == df::unit_labor::PULL_LEVER ||
-                    l == df::unit_labor::HAUL_TRADE);
+                if (Units::isValidLabor((*d)->dwarf, l))
```

Comment:
FYI, you could probably use a range-based for loop here to avoid the double-dereference (`for (auto d : available_dwarfs)`).